utils/log.js

```diff
@@ -28,12 +29,22 @@ exports.errors = new Logging({
   projectId: 'amp-error-reporting',
 }).log('javascript.errors');
 
-exports.users = new Logging({
-  projectId: 'amp-error-reporting-user',
-  keyFilename: 'amp-error-reporting-users.json',
-}).log('javascript.errors');
+keys.users
+  .credentials()
+  .then(credentials => {
+    exports.users = new Logging({
+      projectId: 'amp-error-reporting-user',
+      credentials,
+    }).log('javascript.errors');
+  })
+  .catch(error => console.error(error));
 
-exports.ads = new Logging({
-  projectId: 'amp-error-reporting-ads',
-  keyFilename: 'amp-error-reporting-ads.json',
-}).log('javascript.errors');
+keys.ads
+  .credentials()
+  .then(credentials => {
+    exports.ads = new Logging({
```

Comment:
Export the promise directly. As is, there's now a race condition for when I can access this property.

Reply:
I agree with your comment below, and will simplify to export promises resolving to credentials. Making all of these exports into promises though would complicate all the code that accesses them, and would mean bubbling up `async` or promises everywhere, and makes the testing a lot more obnoxious/fragile cause it requires stubbing out the request before `require`ing anything that imports the log module.

It is indeed a race condition, but in practice, the low latency of Cloud Storage means that this rarely is relevant, especially since with high probability requests are being handled by already-initialized instances. If the handler tries to execute immediately before the keys are fetched, we return a 400 and drop that error, along with millions of others. This feels like a reasonable tradeoff for keeping some of the logic simpler, and we can monitor 400s and adjust if this is a common issue. WDYT?

Reply:
> Making all of these log exports into promises though would complicate all the code that accesses them, and would mean bubbling up async or promises everywhere

We could go back the other direction and continue to use a prestart? That way the keys will be available sync?

In the future, we'll have top level await, which would simplify this greatly.

> cause it requires stubbing out the request before requireing anything that imports the log module

Isn't that the case now?

>  If the handler tries to execute immediately before the keys are fetched, we return a 400 and drop that error, along with millions of others. This feels like a reasonable tradeoff for keeping some of the logic simpler, and we can monitor 400s and adjust if this is a common issue. WDYT?

Why not make the handler wait for the promise to resolve? It's already async, so it shouldn't hurt.